### PR TITLE
Add a search page

### PR DESCRIPTION
**Pull request by michaelenger**
This replaces the "Infinite" link in the header with a search page that lets you search for videos. The search mechanism is a little naïvely implemented, but it works alright for now.

I also took the opportunity to add some unit tests.

Screenshot 2023-09-06 at 21-36-08 Duders Zone

### CODE REVIEW

File: src/routes/search/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		<div id="field-wrapper">
+			<input
+				type="search"
+				id="search"
+				placeholder="search for something"
+				bind:value={searchQuery}
+			/>
+		</div>
+		<div id="button-wrapper">
+			<Button handler={() => {}}>Go Get It</Button>
```
Comment (vegeta897) [inline src/routes/search/+page.svelte:32]: Minor thing, but `() => {}` could be set as a default value for this prop in the Button component so we don't have to include it when we don't need it.

**Review by vegeta897 — APPROVED** [review]
Nice work! I've never looked into how to code a text search so I assume the word indexing of the titles is more efficient than just looking through every full title.

It looks like the results inherit the same sorting as the main video list, which I discovered is not chronological and seems to be just whatever archive.org spits out in the download script. This is fine for now, but maybe chronological would be better.

File: src/routes/search/+page.svelte (hunk 1 above)
Comment (michaelenger) [reply src/routes/search/+page.svelte]: I was considering doing this, but then I ended up not doing it for ..whatever reason? I've changed my mind and made the change.

**Comment by michaelenger** [comment]
@vegeta897 This is a very naïve implementation of a search feature and I already have ideas on how it could be improved. The sorting is one of the things because it doesn't take into consideration if you put in multiple words that appear in a title. It should put the ones with the most matches on top and all that stuff.

I'm probably going to change this soon(ish), but I thought that the work of getting the new page in and such could come first and then we can tweak the actual search logic as we go along.

**Review by vegeta897 — APPROVED** [review]

**Pull request merged**